source/apple/ipc-socket-osx.cpp

```diff
@@ -70,20 +70,24 @@ uint32_t os::apple::socket_osx::read(char *buffer, size_t buffer_length, bool is
     int sizeChunks = 8*1024; // 8KB
     int offset = 0;
     int file_descriptor = t == REQUEST ? file_req : file_rep;
-    std::string typePipe = t == REQUEST ? "request" : "reply";
+    std::string typePipe = t == REQUEST ? "server" : "client";
 
     if (file_descriptor < 0)
         goto end;
 
     while (ret <= 0) {
         ret = ::read(file_descriptor, buffer, buffer_length);
-        while (ret == sizeChunks) {
-            std::cout << "chunk data" << std::endl;
-            offset += sizeChunks;
+        if (ret > 0)
+            offset += ret;
+        while (ret == sizeChunks || offset == sizeChunks) {
             std::vector<char> new_chunks;
             new_chunks.resize(sizeChunks);
             ret = ::read(file_descriptor, new_chunks.data(), new_chunks.size());
-            ::memcpy(&buffer[offset], new_chunks.data(), ret);
+            int errnum;
+            if (ret > 0) {
```

Comment:
if we inside while loop because "offset == sizeChunks" condition. 
and we fail to read we still be in loop till read successful and offset be changed.

Reply:
That's intended, since the read is async, it will return immediately each time we read.